handlers/rsa_test.go

```diff
@@ -0,0 +1,156 @@
+		"efbfbd1731724a2829dcb26fefbfbdefbfbd1125efbfbdefbfbd77efbfbdefbfbd49efbfbd" +
+		"56efbfbd05efbfbdefbfbdefbfbd77efbfbdefbfbd2e64efbfbdefbfbdefbfbdefbfbdefbf" +
+		"bdefbfbdefbfbdefbfbd3f431fefbfbdefbfbdefbfbd241e42efbfbdefbfbdefbfbd35efbf" +
+		"bd555ec3bbd9bc59efbfbdefbfbdefbfbd4cefbfbdefbfbd58e1889defbfbdefbfbd2b13ef" +
+		"bfbdefbfbdefbfbdefbfbd4b45181d78efbfbdefbfbddbabefbfbd7e18efbfbd7fefbfbdef" +
+		"bfbd3cefbfbdefbfbd21efbfbd61efbfbd12efbfbd39efbfbd0aefbfbdefbfbd42417b586f" +
+		"efbfbdefbfbd0b2d6322efbfbd3fefbfbd1a0213efbfbd0653efbfbd78efbfbdefbfbd58ef" +
+		"bfbd2a69c585efbfbdefbfbd61276aefbfbdefbfbd2defbfbdefbfbd091321efbfbdefbfbd" +
+		"efbfbdefbfbdefbfbd0638efbfbd612341efbfbdefbfbd183eefbfbdefbfbdefbfbdefbfbd281e13"
+	rsaPubKey = `-----BEGIN PUBLIC KEY-----
+MIIBIDANBgkqhkiG9w0BAQEFAAOCAQ0AMIIBCAKCAQEApc2hU8zulyJzdQE5IPAv
+B2BgveoZmYUmPEjSb4DViBoATK1hlaY8Psp5vj0H0L4tM8AlXRhPQlECibhgccig
+xQFcG7CLXiSAn7c4XoR+J2SCgx76Fwl9L3WhQigxyKsmpGIqubseydmwfJi4TBnq
+qnX4prsW1PT8GpG35t8Qi8PtkXVGmL7G5pkPXtF0hRzKSfhzsDBbJsl6Jk/Rn5Id
+pKHXL22FdbE9fGzIlW2a6Zdd0b0Q3FZBMnWLSwo0OwBtC/qNnDTCzboig9djiFmA
+yuj8jVhsy050nI72TAONjGKi+xn4lYfdOV2k6TyvpRHfylHouK2v0/bktSlkFI0y
+nwIBAw==
+-----END PUBLIC KEY-----`
+)
+
+func TestPublicKey(t *testing.T) {
+	pubK, err := parseRsaPublicKey(rsaPubKey)
+	qt.Assert(t, err, qt.IsNil)
+	qt.Assert(t, pubK, qt.IsNotNil)
+}
+
+func TestAuthDataParserErr(t *testing.T) {
+	inputVectors := [][]string{
+		{"", "", ""},
+		{"", ""},
+		{""},
+		{"___66cdf1cac93ded9c8d13b7cc74601c7f25bf50d392549e1146eaa8429ab01", voterId, signature},
+		{processId, "___66cdf1cac93ded9c8d13b7cc74601c7f25bf50d392549e1146eaa8429ab01", signature},
+		{processId, voterId, "___66cdf1cac93ded9c8d13b7cc74601c7f25bf50d392549e1146eaa8429ab01"},
+		{"1234", "1234", "1234"},
+	}
+
+	for _, input := range inputVectors {
+		res, err := parseRsaAuthData(input)
+		qt.Assert(t, res, qt.IsNil)
+		qt.Assert(t, err, qt.IsNotNil)
+	}
+}
+
+func TestAuthDataParser(t *testing.T) {
+	res, err := parseRsaAuthData([]string{
+		processId,
+		voterId,
+		signature,
+	})
+
+	qt.Assert(t, res.VoterId, qt.IsNotNil)
+	qt.Assert(t, res.Message, qt.IsNotNil)
+	qt.Assert(t, res.Signature, qt.IsNotNil)
+	qt.Assert(t, err, qt.IsNil)
+
+	aa := hex.EncodeToString(res.VoterId)
+	qt.Assert(t, aa, qt.Equals, voterId)
+
+	bb := hex.EncodeToString(res.Message)
+	qt.Assert(t, bb, qt.Equals, processId+voterId)
+
+	cc := hex.EncodeToString(res.Signature)
+	qt.Assert(t, cc, qt.Equals, signature)
+}
+
+// func TestSignature1(t *testing.T) {
+// 	res, err := parseRsaAuthData([]string{
+// 		processId,
+// 		voterId,
+// 		signature,
+// 	})
+// 	qt.Assert(t, err, qt.IsNil)
+
+// 	pubK, _ := parseRsaPublicKey()
+// 	err = validateRsaSignature(res.Signature, res.Message, pubK)
+
+// 	qt.Assert(t, err, qt.IsNil)
+// }
+
+func TestSignature2(t *testing.T) {
+	// Manual inputs verified on cyberchef
+	message := []byte("hello")
+	sig, _ := hex.DecodeString(
```

Comment:
Please, do not avoid errors on tests, else it will be hard to identify the problem (if any).